### DISCUSSION TITLE
Sustainable API Green Score - potential to donate and work under GSF

### DISCUSSION THREAD

**Discussion by Sealjay** [Green-Software-Foundation/oc#17] [General]
Raised by @jawache and @greenhsu123:

> A request from “Collectif API Thinking” to donate the methodology for calculating the green score methodology to the GSF: API Lifecycle, Data Exchange, Data, Archi, Tools, Infra, Communication.

A suggestion that this could be an open source toolbox, living in Markdown in a Git repository.

More resources:
- YouTube video: https://www.youtube.com/watch?v=5xbmAnoZhLI
- PDF: https://www.collectif-api-thinking.com/assets/deliverables/worksites/50_CAT_API_Sustainable_IT.pdf
- PDF: https://www.collectif-api-thinking.com/assets/deliverables/worksites/48_CAT_Sustainable_API_GreenScore_V1-2.xlsx

**Comment by Sealjay** [comment]
Here are some of my thoughts, using [the maturity model](https://github.com/Green-Software-Foundation/oc/blob/8325ccfcb25849dd1bb6140e7c5373769fe3ddd5/project-lifecycle.md) to guide them a little. 
- The initiative looks great and well thought through, so I'm less concerned from a quality point of view
- Slide 2 of the PDF has a clear description and a well-defined scope - https://www.collectif-api-thinking.com/assets/deliverables/worksites/50_CAT_API_Sustainable_IT.pdf
- If the team contributing the content are willing to be committed development resources, and work with us, as well as initial maintainers, then we have checked that criteria
- As a collective, the initial vendors are not our members, but do appear to be neutral (don't consider this due diligence): https://www.collectif-api-thinking.com/

My main concerns, that we can mitigate by speaking with the group are:
- _Where does this fit with our other work?_ The [identified domains](https://www.collectif-api-thinking.com/assets/deliverables/worksites/50_CAT_API_Sustainable_IT.pdf) (slide 2), don't quite fit with the areas we identified [in the patterns group](https://patterns.greensoftware.foundation/catalog/) - I'm ok with this, as they seem more detailed, but I think we should rationalise both projects to use the same terminology and domains. This applies to other projects too.
- _Where does this fit organisationally? I see this fitting in the Standards WG as a sub-project, if the group are willing to report back. Are the standards WG chairs willing to take this on?
- _What's the long term plan for this?_ The evaluation grid is likely something new - and we need a set of heuristic rules, where you could select a domain area, and answer yes/no questions, to identify relevant patterns. However, the rules like could end up in the patterns catalogue - **are the collective donating the code happy with this as an approach?**

**Reply by jawache** [reply]
Some of those were my first thoughts:

- We don't have a mechanism for non-members leading projects in the GSF so I suppose the biggest step would be to find someone who's interested enough in this to drive it internally (or ask them to join and have them lead it as members).
- We can provide PM support but it's going to need a lead who can commit a few hours a week at least to work on it.
- It might fit into Standard WG, or Policy WG does have a provision for a "green product certification" in it's scope definition (cc/ @mrchrisadams @elisezelechowski) and this would be the first step in any certification.
- I would say it has to be marked as _incubating_ until there is a release (that's a test also, if it doesn't quite gain the momentum of interest in GSF members to get it over that line then at least it stays as incubated and we don't set an expectation of support/ownership)


- Agreed with your points on what's the long term plan here. Do we want to create something similar but for all software products? 
- How does it match with the SCI? I know we've had questions of making the SCI give an A,B,C,D score to things and it's never been of much interest.
- I'm personally hesitant, A- D scores feels pretty arbitrary and subjective and I prefer to avoid decisions that are subjective since then most of your time is spent explaining to someone who's upset they scored a D why it's right they scored a D. I like the SCI since it's more objective, it's an equation, don't blame my categorization of patterns, argue the math's of an equation.

**Comment by tmcclell** [comment]
Discussions of methodologies for quantifying data points and range for scoring along with a checklist. @greenhsu123 - direction to sponsors of API donation? @seanmcilroy29 - questions about oversight of project. @jawache - generate list of issues and concerns to have the initial discussion. @jawache - taking a list of the questions and discuss at the next meeting.

**Comment by Sealjay** [comment]
Propose we talk about "code acceptance" rather than donation @Jawache - we don't want to accept everything, it has to be good quality :)

**Comment by tmcclell** [comment]
@jawache Need to be a member company. If not a member, give up governance of the project and working group. Disagreements deferred to working groups for decisions. @jawache to summarize donation process of our website and discuss with donators.

**Comment by tmcclell** [comment]
Deferment until next meeting.

**Comment by jawache** [comment]
Seeing this linked again so locking this conversation, this was actioned and the content on the website updated: https://oc.greensoftware.foundation/project-donations.html
